nixos/modules/services/web-servers/nginx/default.nix

```
@@ -402,24 +402,63 @@ let
     -subj "/C=UK/ST=Warwickshire/L=Leamington/O=OrgName/OU=IT Department/CN=example.com"
     openssl x509 -req -days 1 -in server.csr -signkey $out/server.key -out $out/server.crt
   '';
-  validatedConfigFile = pkgs.runCommand "validated-nginx.conf" { nativeBuildInputs = [ cfg.package ]; } ''
+  fakednsConfig = builtins.toFile "fakednsConfig" ''
+    [Settings]
+    ListenOnPort = 53
+    [DomainPattern]
+    name_pattern = *
+    answer_A = 1.2.3.4
+    answer_AAAA = 8::1
+  '';
+  nginxWithResolver = pkgs.writeShellScript "nginxWithResolver" ''
+    set -e
+    fakedns --config ${fakednsConfig} &
+    # wait for the server to be ready
+    for i in $(seq 1 1000); do
```

```suggestion
    for i in $(seq 1 30); do
```
waiting 1000 seconds for that to come up is to long